crypto/threads_win.c

```diff
@@ -39,18 +39,27 @@ CRYPTO_RWLOCK *CRYPTO_THREAD_lock_new(void)
 
 int CRYPTO_THREAD_read_lock(CRYPTO_RWLOCK *lock)
 {
+    if (NULL == lock)
+        return 0;
```

Comment:
Looking at the callstack in the screenshot, it is clear that it's the `rand_meth_lock` which is uninitialized. However, this  lock should be initialized at first call via `do_rand_init()` and then be valid until ` rand_cleanup_int()` is called at program exit. If this pointer is NULL then something has fundamentally gone wrong and the reason needs to be found and fixed instead.

https://github.com/openssl/openssl/blob/e9f148c9356b18995298f37bafbf1836a3fce078/crypto/rand/rand_lib.c#L726-L733